File: cpp/example/src/FindNuniqueIntSum2Zero/findNuniqueIntSum2Zero.cpp
#include "findNuniqueIntSum2Zero.h"

vector<int> findNuniqueIntSum2Zero::naive(int n){
    vector<int> result = {};
    if (n%2==1) result.push_back(0);
    for (int i=0;i<n/2;i++){
        int tmp = i+1;
        result.push_back(tmp);
        result.push_back(-tmp);
    }
    return result;
}

File: cs/sample/Sample/FindNuniqueIntSum2Zero/FindNuniqueIntSum2Zero.cs
using System;
using System.Collections.Generic;

namespace Sample.FindNuniqueIntSum2Zero{
    public class findNuniqueIntSum2Zero{
        public static int[] naive(int n){
            int[] result = new int[n];
            int i = 0, j = 0;
            if (n%2==1) 
                result[i] = j;
            i+=1;j+=1;
            while (i<n){
                result[i] = j;
                result[i+1] = -j;
                i+=2;j+=1;
            }
            return result;
        }
    }
}
